src/lib.rs

```diff
@@ -192,25 +195,21 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
-    use futures::future;
-    use futures::{stream, FutureExt, StreamExt, TryFutureExt};
+    use futures::{stream, FutureExt, StreamExt};
     use std::iter;
     use std::time::{Duration, Instant};
+    use tokio::runtime::Runtime;
 
     #[test]
     fn messages_pass_through() {
         let v = stream::iter(iter::once(5))
             .chunks_timeout(5, Duration::new(1, 0))
             .collect::<Vec<_>>();
-        tokio::run(
-            v.then(|x| {
-                assert_eq!(vec![vec![5]], x);
-                future::ready(())
-            })
-            .unit_error()
-            .boxed()
-            .compat(),
-        );
+
+        let mut rt = Runtime::new().expect("Failed to create tokio runtime");
+        rt.block_on(async move {
```

Comment:
you could use `tokio::spawn_blocking` for the tests (https://docs.rs/tokio/0.2.2/tokio/task/fn.spawn_blocking.html) or even better: use the `#[tokio::test]` macro

Reply:
Thanks! I wasn't aware that even existed